src/goto-programs/validate_goto_model.h

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************\
+Module: Validate Goto Programs
+
+Author: Diffblue
+
+Date: Oct 2018
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_PROGRAMS_VALIDATE_GOTO_MODEL_H
+#define CPROVER_GOTO_PROGRAMS_VALIDATE_GOTO_MODEL_H
+
+#include <util/validate.h>
+//#include "goto_functions.h"
+
+struct goto_model_validation_optionst
+{
+  bool entry_point_exists;
+  bool function_pointer_calls_removed;
+  bool check_returns_removed;
+  bool check_called_functions;
+  bool check_source_location;
+
+  goto_model_validation_optionst()
+    : entry_point_exists{true},
+      function_pointer_calls_removed{false},
+      check_returns_removed{false},
+      check_called_functions{false},
+      check_source_location{false}
+  {
+  }
+
+  explicit goto_model_validation_optionst(bool options_value)
```

Comment:
Agree, please don't use Booleans without further comment. I think in this case a comment may be good enough, but otherwise the options @hannes-steffenhagen-diffblue also seem good to me.

Reply:
Addressed via nested enum class.